File: dotnet/src/BullMQ/JobState.cs
namespace BullMQ;

/// <summary>
/// The lifecycle states a job can be in. The string values match the Redis
/// key/list names used by the shared Lua scripts.
/// </summary>
public enum JobState
{
    Completed,
    Failed,
    Delayed,
    Active,
    Prioritized,
    Waiting,
    WaitingChildren,
    Unknown,
}

/// <summary>Helpers for converting <see cref="JobState"/> to/from its wire string.</summary>
public static class JobStateExtensions
{
    public static string ToWireString(this JobState state) => state switch
    {
        JobState.Completed => "completed",
        JobState.Failed => "failed",
        JobState.Delayed => "delayed",
        JobState.Active => "active",
        JobState.Prioritized => "prioritized",
        JobState.Waiting => "waiting",
        JobState.WaitingChildren => "waiting-children",
        _ => "unknown",
    };

    public static JobState FromWireString(string? value) => value switch
    {
        "completed" => JobState.Completed,
        "failed" => JobState.Failed,
        "delayed" => JobState.Delayed,
        "active" => JobState.Active,
        "prioritized" => JobState.Prioritized,
        "waiting" => JobState.Waiting,
        "waiting-children" => JobState.WaitingChildren,
        _ => JobState.Unknown,
    };
}

File: java/src/main/java/io/bullmq/core/JobState.java
package io.bullmq.core;

/**
 * The lifecycle states a job can be in. The string values match the Redis
 * key/list names used by the shared Lua scripts.
 */
public enum JobState {
    COMPLETED,
    FAILED,
    DELAYED,
    ACTIVE,
    PRIORITIZED,
    WAITING,
    WAITING_CHILDREN,
    UNKNOWN
}

/**
 * Helpers for converting {@link JobState} to/from its wire string.
 */
class JobStateExtensions { // Made package-private

    JobStateExtensions() {
        // Utility class - prevent instantiation
    }

    static String toWireString(JobState state) {
        return switch (state) {
            case COMPLETED -> "completed";
            case FAILED -> "failed";
            case DELAYED -> "delayed";
            case ACTIVE -> "active";
            case PRIORITIZED -> "prioritized";
            case WAITING -> "waiting";
            case WAITING_CHILDREN -> "waiting-children";
            case UNKNOWN -> "unknown";
        };
    }

    static JobState fromWireString(String value) {
        return switch (value) {
            case "completed" -> JobState.COMPLETED;
            case "failed" -> JobState.FAILED;
            case "delayed" -> JobState.DELAYED;
            case "active" -> JobState.ACTIVE;
            case "prioritized" -> JobState.PRIORITIZED;
            case "waiting" -> JobState.WAITING;
            case "waiting-children" -> JobState.WAITING_CHILDREN;
            default -> JobState.UNKNOWN;
        };
    }
}